federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/membership/group/GroupLDAPStorageMapper.java

```diff
@@ -361,22 +363,42 @@ private void updateAttributesOfKCGroup(GroupModel kcGroup, LDAPObject ldapGroup)
     }
 
 
-    protected GroupModel findKcGroupByLDAPGroup(RealmModel realm, LDAPObject ldapGroup) {
+    protected GroupModel findKcGroupByLDAPGroup(RealmModel realm, GroupModel parent, LDAPObject ldapGroup) {
         String groupNameAttr = config.getGroupNameLdapAttribute();
         String groupName = ldapGroup.getAttributeAsString(groupNameAttr);
 
         if (config.isPreserveGroupsInheritance()) {
             // Override if better effectivity or different algorithm is needed
-            return getAllKcGroups(realm)
+            return getAllKcGroups(realm, parent)
                     .filter(group -> Objects.equals(group.getName(), groupName)).findFirst().orElse(null);
         } else {
             // Without preserved inheritance, it's always at groups path
-            return KeycloakModelUtils.findGroupByPath(realm, getKcGroupPathFromLDAPGroupName(groupName));
+            return realm.getGroupByName(groupName, parent);
+        }
+    }
+    // TODO how this will work with group names with a / without hitting the db everytime
+    private GroupModel findGroupByPath(RealmModel realm, String path) {
+        if (path == null) {
+            return null;
+        }
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        if (path.endsWith("/")) {
+            path = path.substring(0, path.length() - 1);
+        }
+        String[] split = path.split("/");
```

Comment:
Minor: It will be good if calls, which are called often and affect the performance use the regex pattern instead of directly calling `String.split` . I am pretty sure you remember your recent PR for the LdapDn performance improvement, which was related exactly to this :-)

Reply:
If the pattern is short there is a fast path in `String.split` without a Pattern. I need to check if a static Pattern will improve performance in this case.